app/controllers/update_game.rb

```diff
@@ -1,23 +1,31 @@
+class UpdateGame
+  def update(current_game, position)
+    board = current_game.board
+    player = current_game.player
+    opponent = current_game.opponent
 
-  class UpdateGame
-    def update(current_game,position)
-      board = current_game.board
-      player = current_game.player
-      opponent = current_game.opponent
-      @current_player = current_game.current_player
+    toggle = TicTacToeGame::Toggle.new(player)
+    @game = TicTacToeGame::Game.new(board, player, toggle)
 
-      toggle = TicTacToeGame::Toggle.new(player)
-      @game = TicTacToeGame::Game.new(board, player, toggle)
-  
-      return false unless board[position] == "-" 
-      GameMove.new.all_moves(@game, opponent, @current_player, position, current_game)
-      current_game.player = toggle.current_turn
-      current_game.board = @game.board
-    end
-    
-    def output(current_game, position)
-      update(current_game, position)
-      game_over = GameoverType.new
-      game_over.message(@game,@current_player) 
+    return false unless board[position] == "-"
+    game_move = GameMove.new
+    current_player_move(current_game, game_move, opponent, position, toggle)
+
+    if !@game.end? && current_game.current_player == "computer"
+      current_player_move(current_game, game_move, opponent, position, toggle)
```

Comment:
This is a nice extract method! Can we take it one step further and turn it into a while loop? So we keep playing the game until either the game is over or the current player is not a computer.